platforms/software/dependency-management/src/main/java/org/gradle/internal/component/AmbiguousVariantSelectionException.java

```diff
@@ -20,15 +20,17 @@
  * This type is {@code deprecated} and will be removed in Gradle 9.0.
  *
  * This is temporarily available for migration only.
- * This exception class is internal and has been replaced by {@link AmbiguousArtifactVariantsException}, which is also internal. 
+ * This exception class is internal and has been replaced by {@link AmbiguousArtifactVariantsException}, which is also internal.
  * If possible, catch a {@link RuntimeException} instead to avoid depending on Gradle internal classes.
  */
+@SuppressWarnings("DeprecatedIsStillUsed")
 @Deprecated
 public abstract class AmbiguousVariantSelectionException extends ArtifactVariantSelectionException {
     public AmbiguousVariantSelectionException(String message) {
         super(message);
     }
 
+    @SuppressWarnings("unused")
     public AmbiguousVariantSelectionException(String message, Throwable cause) {
```

Comment:
Nobody outside of Gradle should be _constructing_ this exception. If we don't use this constructor I think we can just delete it